gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Subscription.java

```diff
@@ -103,62 +131,172 @@ public Builder toBuilder() {
+   * Deletes this subscription.
+   *
+   * @return {@code true} if the subscription was deleted, {@code false} if it was not found
+   * @throws PubSubException upon failure
+   */
   public boolean delete() {
     return pubsub.deleteSubscription(name());
   }
 
+  /**
+   * Sends a request for deleting this subscription. This method returns a {@code Future} object to
+   * consume the result. {@link Future#get()} returns {@code true} if the subscription was deleted,
+   * {@code false} if it was not found.
+   */
   public Future<Boolean> deleteAsync() {
     return pubsub.deleteSubscriptionAsync(name());
   }
 
+  /**
+   * Fetches current subscription's latest information. Returns {@code null} if the subscription
+   * does not exist.
+   *
+   * @return a {@code Subscription} object with latest information or {@code null} if not found
+   * @throws PubSubException upon failure
+   */
   public Subscription reload() {
     return pubsub.getSubscription(name());
   }
 
+  /**
+   * Sends a request for fetching current subscription's latest information. This method returns a
+   * {@code Future} object to consume the result. {@link Future#get()} returns the requested
+   * subscription or {@code null} if not found.
+   *
+   * @return a {@code Subscription} object with latest information or {@code null} if not found
+   * @throws PubSubException upon failure
+   */
   public Future<Subscription> reloadAsync() {
     return pubsub.getSubscriptionAsync(name());
   }
 
+  /**
+   * Sets the push configuration for this subscription. This may be used to change a push
+   * subscription to a pull one (passing a {@code null} {@code pushConfig} parameter) or vice versa.
+   * This methods can also be used to change the endpoint URL and other attributes of a push
+   * subscription. Messages will accumulate for delivery regardless of changes to the push
+   * configuration.
+   *
+   * @param pushConfig the new push configuration. Use {@code null} to unset it
+   * @throws PubSubException upon failure, or if the subscription does not exist
+   */
   public void replacePushConfig(PushConfig pushConfig) {
     pubsub.replacePushConfig(name(), pushConfig);
   }
 
+  /**
+   * Sends a request for updating the push configuration for a specified subscription. This may be
+   * used to change a push subscription to a pull one (passing a {@code null} {@code pushConfig}
+   * parameter) or vice versa. This methods can also be used to change the endpoint URL and other
+   * attributes of a push subscription. Messages will accumulate for delivery regardless of changes
+   * to the push configuration. The method returns a {@code Future} object that can be used to wait
+   * for the replace operation to be completed.
+   *
+   * @param pushConfig the new push configuration. Use {@code null} to unset it
+   * @return a {@code Future} to wait for the replace operation to be completed.
+   */
   public Future<Void> replacePushConfigAsync(PushConfig pushConfig) {
```

Comment:
Yep it's void. If it returns successfully the change was applied otherwise you get an exception.